core/app/models/spree/shipping_method.rb

```diff
@@ -20,6 +20,8 @@ class ShippingMethod < Spree::Base
                                     :class_name => 'Spree::Zone',
                                     :foreign_key => 'shipping_method_id'
 
+    has_and_belongs_to_many :stock_locations
```

Comment:
The `#has_and_belongs_to_many` is discouraged in Rails now. Can you make this a `#has_many :though` association?